### PR TITLE
VACMS-126703: Outdated Content Notifications

### LINKED ISSUE #12703: Send Outdated Content email notification for testing
## Description
Theme and send email notifications for outdated content.

## Acceptance Criteria
- [x] Build template of email (pending final designs)
- [x] Build queue worker to send email runs on cron, batch at 10
- [x] Add error reporting if queue is not getting smaller (see migration or govdelivery batch)

### Team
Please check the team(s) that will do this work.

- [x] `CMS Team`
- [ ] `Public Websites`
- [ ] `Facilities`
- [ ] `User support`

### LINKED ISSUE #12711: Smoke test distribution of CMS Notification Emails on Tugboat
## Description
We want to ensure that CMS notification emails work as expected. We would like to see what Mail Hog catches/shows before we send out the emails directly to editors. 

Mail Hog will allow us to see the different pieces of the 6102 emails are there and they are sending the correct info.

## Acceptance Criteria

- [ ] create tugboat with database export that includes email addresses (not the standard one) 
- [ ] Manually kick off service and monitor the queue 
- [ ] confirm that the process works as expected
- [ ] Design and accessibility review completed
- include Laura and Blake in the test email

### PR DESCRIPTION
**Pull request by edmund-dunn**
## Description

Closes #12703
Closes #12711

## Testing done
Manually ran job using `drush php-eval "print_r(\Drupal::service('va_gov_notifications.outdated_content')->checkForOutdatedContent());"` to queue up notifications, ran cron from `/admin/config/system/cron` to process queue and verified emails in mailhog. 

## QA steps

What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

As engineer
- [ ] Shell into tugboat
- [ ] Run `drush php-eval "print_r(\Drupal::service('va_gov_notifications.outdated_content')->checkForOutdatedContent());"`
- [ ] Visit `/admin/config/system/queues` to verify jobs are queued
- [ ] Go to `/admin/config/system/cron` and run cron
- [ ] re-Visit `/admin/config/system/queues` to verify jobs processing successfully
- [ ] Check the Captured Mail section of the individual Tugboat to see if email were sent, and to see the actual emails

As Designer/a11y: you should have received a copy of the email in your va.gov email
- [ ] Email meets design 
- [ ] Email passes a11y

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [x] `CMS Team`
- [ ] `Public websites`
- [ ] `Facilities`
- [ ] `User support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by laflannery** [comment]
**Comments and requests:**
- Would it be possible to set a hover/visited color for the Review link, after I click the link it's blue and very hard to see on the red.  Can we keep it white?
- When I click the 6102 link in the top section of the email it never completes, just loads and loads forever.  I think the link is wrong, can you change it to: https://www.va.gov/web/index.cfm
- When I click review, I am not prompted to login but instead I am brought directly to an outdated content report with no sections and no results: This seems not right, but I'm not sure if this is because we are just testing?
image

**Comment by EWashb** [comment]
@edmund-dunn We've had some changes to the text of the email per feedback from the PO. Here's a screenshot, but also including the [figma link](https://www.figma.com/file/fQl2Y2k9MDA2dc0xViC645/6102-Notification-Email?node-id=314%3A6147&t=s8EVmgrKuQtWZDch-1). Please note that the "directions" header has been removed. We tried to make the text more generic regardless of editor type. Also, we are including a link to the KB article

Screenshot 2023-03-03 at 3 49 08 PM

**Review by laflannery — DISMISSED** [review]

File: docroot/modules/custom/va_gov_notifications/va_gov_notifications.module [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  // Any fields on the message can be made available as template variables.
+  $message = $variables['message'];
+  $module_handler = \Drupal::service('module_handler');
+  $path = $module_handler->getModule('va_gov_notifications')->getPath();
+  $variables['user_name'] = $message->get('field_editor_username')->getString();
+  $variables['banner'] = '/' . $path . '/assets/outdated-content-email-banner.png';
+}
+
+/**
+ *
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_notifications/va_gov_notifications.module:35]:
```suggestion
 * Implements hook_mail_alter().
```

Comment (edmund-dunn) [reply docroot/modules/custom/va_gov_notifications/va_gov_notifications.module:35]: Good catch!

**Comment by ndouglas** [comment]
The code looks great.

I ran the command to load the queue twice, and it, uh, it loads the queue twice:

Screenshot 2023-03-07 at 3 11 09 PM

Thinking about a worst case scenario, some dumb glitch or something like a cron run doesn't complete successfully and update the proper timestamp, etc etc etc.

Are these emails coalesced?  Is it worth considering something that only notifies if there's not a queued notification already?  Is it worth erroring out if we attempt to load an already-loaded queue?  Am I just being paranoid?

**Review by ndouglas — DISMISSED** [review]

**Review by BlakeOrgan — DISMISSED** [review]

**Pull request merged**